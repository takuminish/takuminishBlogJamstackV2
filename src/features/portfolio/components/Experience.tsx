import React from "react";
import Timeline from "@mui/lab/Timeline";
import TimelineItem from "@mui/lab/TimelineItem";
import TimelineSeparator from "@mui/lab/TimelineSeparator";
import TimelineConnector from "@mui/lab/TimelineConnector";
import TimelineContent from "@mui/lab/TimelineContent";
import TimelineDot from "@mui/lab/TimelineDot";
import { format } from "date-fns";
import { Experience } from "../../../pages";

type Props = {
  experiences: Experience[];
};

function History(props: Props) {
  const { experiences } = props;

  return (
    <Timeline position="alternate">
      {experiences.map((experience) => (
        <TimelineItem key={experience.title}>
          <TimelineSeparator
            sx={{
              margin: "0 7px",
            }}
          >
            <TimelineDot variant="outlined" color="success" />
            <TimelineConnector sx={{ backgroundColor: "#2f7d31" }} />
          </TimelineSeparator>
          <TimelineContent
            sx={{
              boxShadow: "1px 1px 1px 1px #dddddd",
              borderRadius: "10px",
              height: "100%",
            }}
          >
            {experience.title}{" "}
            {format(new Date(experience.startAt), "yyyy/MM/dd")} ~{" "}
            {experience.endAt
              ? format(new Date(experience.endAt), "yyyy/MM/dd")
              : ""}
          </TimelineContent>
        </TimelineItem>
      ))}
    </Timeline>
  );
}

export default History;
