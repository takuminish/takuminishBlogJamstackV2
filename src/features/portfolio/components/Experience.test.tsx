import React from "react";
import { render } from "@testing-library/react";
import Experience, { Props } from "./Experience";

const createDOM = () => {
  const props: Props = {
    experiences: [
      {
        fieldId: "experience",
        title: "experience1",
        startAt: "2015-03-31T15:00:00.000Z",
        endAt: "2019-03-30T15:00:00.000Z",
        detailedInformation: "<p> </p>",
      },
      {
        fieldId: "experience",
        title: "experience2",
        startAt: "2015-03-31T15:00:00.000Z",
        endAt: "2019-03-30T15:00:00.000Z",
        detailedInformation: "<p> </p>",
      },
    ],
  };
  return render(<Experience experiences={props.experiences} />);
};

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});
