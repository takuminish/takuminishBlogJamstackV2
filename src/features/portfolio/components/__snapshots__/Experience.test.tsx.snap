// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <ul
    class="MuiTimeline-root MuiTimeline-positionAlternate css-55vwbm-MuiTimeline-root"
  >
    <li
      class="MuiTimelineItem-root MuiTimelineItem-positionAlternate MuiTimelineItem-missingOppositeContent css-1v6o2d9-MuiTimelineItem-root"
    >
      <div
        class="MuiTimelineSeparator-root css-jw6nla-MuiTimelineSeparator-root"
      >
        <span
          class="MuiTimelineDot-root MuiTimelineDot-outlined MuiTimelineDot-outlinedSuccess css-sk7zve-MuiTimelineDot-root"
        />
        <span
          class="MuiTimelineConnector-root css-1kdfyxn-MuiTimelineConnector-root"
        />
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionAlternate css-1vbfgko-MuiTypography-root-MuiTimelineContent-root"
      >
        experience1
         
        2015/04/01
         ~
         
        2019/03/31
      </div>
    </li>
    <li
      class="MuiTimelineItem-root MuiTimelineItem-positionAlternate MuiTimelineItem-missingOppositeContent css-1v6o2d9-MuiTimelineItem-root"
    >
      <div
        class="MuiTimelineSeparator-root css-jw6nla-MuiTimelineSeparator-root"
      >
        <span
          class="MuiTimelineDot-root MuiTimelineDot-outlined MuiTimelineDot-outlinedSuccess css-sk7zve-MuiTimelineDot-root"
        />
        <span
          class="MuiTimelineConnector-root css-1kdfyxn-MuiTimelineConnector-root"
        />
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionAlternate css-1vbfgko-MuiTypography-root-MuiTimelineContent-root"
      >
        experience2
         
        2015/04/01
         ~
         
        2019/03/31
      </div>
    </li>
  </ul>
</div>
`;
